### PR TITLE
Added slug to the Frontmatter Reference in the Starlight Docs

### PR DESCRIPTION
**Pull request by chrskerr**
# Added slug to the Frontmatter Reference in the Starlight Docs

## What does this PR change? Give us a brief description.

I recently set up a starlight page, and needed to unify my slug generation to make some pre-processing easier (I was pulling my content from Notion via a build-script 😅). Through the Astro docs ([ref](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs)) I realised that setting the `slug` field in the frontmatter for a page will override the default slug.

Since this was hard to find, I am submitting this PR to add the instructions to the Starlight docs in case anyone else needs it.

## Screenshot
Screenshot 2023-12-10 at 11 35 42 am

### CODE REVIEW

File: docs/src/content/docs/reference/frontmatter.md [hunk 1]
```diff
@@ -29,6 +29,12 @@ You must provide a title for every page. This will be displayed at the top of th
 
 The page description is used for page metadata and will be picked up by search engines and in social media previews.
 
+### `slug`
+
+**type**: `string`
+
+Overrides the generated slug for the page.
```
Comment (kevinzunigacuellar) [inline docs/src/content/docs/reference/frontmatter.md:36]: I think we can mention that this property comes from content collections and its not a Starlight specific, this way we could include a link back to the official docs for more information. Let me know what you think.

```suggestion
Content collections reserved property to override the slug of the page, check [defining custom slugs](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) for more details.
```

Comment (chrskerr-canva) [reply docs/src/content/docs/reference/frontmatter.md:36]: Sounds great to me, thank you!

Comment (chrskerr) [reply docs/src/content/docs/reference/frontmatter.md:36]: Sounds great to me, thanks!

**Review by ElianCodes — COMMENTED** [review]
Yeah, I guess it makes sense to add this for people who never used Astro and want to understand the behaviour of content collections.

File: docs/src/content/docs/reference/frontmatter.md [hunk 2]
```diff
@@ -29,6 +29,12 @@ You must provide a title for every page. This will be displayed at the top of th
 
 The page description is used for page metadata and will be picked up by search engines and in social media previews.
 
+### `slug`
+
+**type**: `string`
+
+Content collections reserved property to override the slug of the page, check [defining custom slugs](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) for more details.
```
Comment (delucis) [inline docs/src/content/docs/reference/frontmatter.md:36]: Thanks for the suggestion @chrskerr! In general we try not to document too much that’s part of Astro’s default functionality, but I agree in this case it’s helpful. Going to make one slight tweak to the proposed text and then should be good to merge.

```suggestion
Override the slug of the page, see [“Defining custom slugs”](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs) in the Astro docs for more details.
```

**Review by delucis — APPROVED** [review]

**Pull request merged**